### PR TITLE
Re-enable Ed25519 functionality with pynacl...

### PR DESCRIPTION
**Pull request by MHHukiewitz**
...as it is already being installed with aleph-client (instead of pysodium)

### CODE REVIEW

File: src/aleph_pytezos/crypto/key.py [hunk 1]
```diff
@@ -126,12 +129,13 @@ def from_secret_exponent(
...
         # Ed25519
         if curve == b'ed':
-            raise Exception("Curve ed25519 not supported.")
             # Dealing with secret exponent or seed?
-            #if len(secret_exponent) == 64:
-            #    public_point = pysodium.crypto_sign_sk_to_pk(sk=secret_exponent)
-            #else:
-            #    public_point, secret_exponent = pysodium.crypto_sign_seed_keypair(seed=secret_exponent)
+            if len(secret_exponent) == 64:
+                public_point = nacl.signing.SigningKey(secret_exponent[32:]).verify_key.encode()
```
Comment (hoh) [inline src/aleph_pytezos/crypto/key.py]: Why this `[32:]` ? Can this be documented in a comment ?

File: src/aleph_pytezos/crypto/key.py [hunk 2]
```diff
@@ -126,12 +129,13 @@ def from_secret_exponent(
...
+            else:
```
Comment (hoh) [inline src/aleph_pytezos/crypto/key.py:138]: Should we `assert` another length here ?

Comment (MHHukiewitz) [reply src/aleph_pytezos/crypto/key.py:138]: The standard format of a secret key is 64 bytes, the first 32 bytes are the public key, the last 32 are secret.

File: src/aleph_pytezos/crypto/key.py (hunk 1 above)
Comment (MHHukiewitz) [reply src/aleph_pytezos/crypto/key.py]: See below, it is how pynacl treats Ed25519 secret keys. When generating a secret key for i.e. Solana, you will get the same format of 64 bytes.

Comment (MHHukiewitz) [reply src/aleph_pytezos/crypto/key.py]: But, to be honest, it could be nicer if
`nacl.signer.VerifyKey(secret_exponent[:32]).encode()`

Comment (hoh) [reply src/aleph_pytezos/crypto/key.py]: Would something like this be more explicit then ?

```suggestion
                # The standard format of a secret key is 64 bytes, the first 32 bytes are the public key, the last 32 are secret.
                public_key, secret = secret_exponent[:32], secret_exponent[32:]
                public_point = nacl.signing.SigningKey(secret).verify_key.encode()
```

File: src/aleph_pytezos/crypto/key.py (hunk 2 above)
Comment (MHHukiewitz) [reply src/aleph_pytezos/crypto/key.py:138]: Ah, you're right. I'll add some additional checks

File: src/aleph_pytezos/crypto/key.py [hunk 3]
```diff
@@ -126,12 +129,17 @@ def from_secret_exponent(
...
+            if len(secret_exponent) == 64:
+                # The standard format of a secret key is 64 bytes,
+                # the first 32 bytes are the public key, the last 32 are secret (or seed).
+                public_key, secret_exponent = secret_exponent[:32], secret_exponent[32:]
+                public_point = nacl.signing.SigningKey(secret_exponent).verify_key.encode()
+            elif len(secret_exponent) == 32:
+                keypair = nacl.signing.SigningKey(secret_exponent)
+                public_point = keypair.verify_key.encode()
+            else:
+                raise AssertionError()
```
Comment (hoh) [inline src/aleph_pytezos/crypto/key.py]: That is not a very Pythonic way of writing it, the following would be more clear.

```suggestion
            else:
                assert len(secret_exponent) == 32, f"Secret exponent length must be 32 or 64, not {len(secret_exponent)}"
                keypair = nacl.signing.SigningKey(secret_exponent)
                public_point = keypair.verify_key.encode()
```

**Review by hoh — COMMENTED** [review]
Do we want to add unit tests on this PR or later ?

**Comment by MHHukiewitz** [comment]
> Do we want to add unit tests on this PR or later ?

It's dubious to me, whether we will keep on using a custom version of pytezos, so for now I would be fine with only having unit tests in aleph-client's Tezos implementation.

**Pull request merged**